### DISCUSSION TITLE
Authentication

### DISCUSSION THREAD

**Discussion by jbouder** [harlan-zw/unlighthouse#171] [Q&A]
I'm trying to figure out the best way to authenticate to my app. I've tried a few of the methods here: https://unlighthouse.dev/guide/guides/authentication, with no luck. Right now just playing with basic auth. My first attempt was to configure auth in my config like here: 
```
export default {
  auth: {
    username: 'username',
    password: 'password',
  },
}
```

My expectation was that unlighthouse would attempt to enter these values into a login screen if encountered...if that's not true, definitely would like some validation there...and if not, how should I use these values?

I also tried to get Programatic Usage working, but I'm not having any luck there...plus as it is experimental, i'm trying to avoid it. TIA!

**Comment by jbouder** [comment]
Nm, I dug into the source and realized I was off base.

**Reply by harlan-zw** [reply]
Glad you could solve it :)
